### PR TITLE
Start converting commands to use argparse.

### PR DESCRIPTION
**Pull request by mikebryant**

### CODE REVIEW

**Comment by Thynix** [comment]
Excellent! Thank you for doing this. I'm [busy this weekend](http://www.wolverinesoft.org/event/contest/space_jam_2015/) and will get to this on Monday at the earliest.

**Comment by Thynix** [comment]
I haven't looked at the content yet, but once that's sorted out could you please also remove the full stops from the end of the commit summaries?

File: fcp/arguments.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        raise argparse.ArgumentTypeError("Invalid timeout: %s" % time_str)
+
+def add_default_arguments(parser):
+    '''
+    Add the default arguments to an argparse parser.
+    '''
+    parser.add_argument('--version', '-V', action='version', version="%(prog)s " + fcp.node.fcpVersion)
+    parser.add_argument('--fcphost', '-H', default=fcp.node.defaultFCPHost, help='Connect to FCP service at host <FCPHOST>. (You may also use the environment variable FCP_HOST)')
+    parser.add_argument('--fcpport', '-P', default=fcp.node.defaultFCPPort, type=int, help='Connect to FCP service at port <FCPPORT>. (You may also use the environment variable FCP_PORT)')
+    parser.add_argument('--timeout', '-t', default=fcp.node.ONE_YEAR, type=timeout_type, help='Set the timeout, in seconds, for completion. Default one year')
```
Comment (Thynix) [inline fcp/arguments.py]: These lines seem rather long - they're a tight fit at least on my monitor - and [PEP8 suggests](https://www.python.org/dev/peps/pep-0008/#maximum-line-length) a line length limit of 79 characters. What do you think of 5fef46d91941ce7ed8569e707786528e00ba6c9d? It also adds full stops, which is arguably poor practice on my part, but anyway argparse removes extra whitespace so it does not add gaps to the `--help` output.

**Comment by Thynix** [comment]
Other than that these changes look fantastic and I will be happy to merge them.

**Comment by Thynix** [comment]
Any word on this?

**Comment by mikebryant** [comment]
Sorry, I was busy when I got the previous comments and then it completely dropped off my radar.

What do you think of that style? When splitting lines I've always hated having varying numbers of parameters in each line, especially for later changes where adding a parameter means shuffling them all around to fit.

**Comment by Thynix** [comment]
That looks just fine. Thanks!

**Pull request merged**